src/Shared/FileUtilities.cs

```diff
@@ -485,6 +485,42 @@ internal static string MaybeAdjustFilePath(string value, string baseDirectory =
             return shouldAdjust ? newValue.ToString() : value;
         }
 
+#if !CLR2COMPATIBILITY
+        /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// This overload takes and returns ReadOnlyMemory of characters.
+        /// </summary>
+        internal static ReadOnlyMemory<char> MaybeAdjustFilePath(ReadOnlyMemory<char> value, string baseDirectory = "")
+        {
+            if (NativeMethodsShared.IsWindows || value.IsEmpty)
+            {
+                return value;
+            }
+
+            // Don't bother with arrays or properties or network paths.
+            if (value.Length >= 2)
+            {
+                var span = value.Span;
+                if ((span[1] == '(' && (span[0] == '$' || span[0] == '@')) ||
+                    (span[1] == '\\' && span[0] == '\\'))
+                {
+                    return value;
+                }
+            }
+
+            // For Unix-like systems, we may want to convert backslashes to slashes
+#if FEATURE_SPAN
```

Comment:
I thought you were eliminating FEATURE_SPAN? And this is under the CLR2COMPATIBILITY check.

Reply:
That's correct. If #6130 is merged first, I'll tweak this PR to remove the ifdef. And vice versa.

Reply:
I have rebased this PR on top of current master and resolved conflicts.